### PR TITLE
Refactor using of crypto functions

### PR DESCRIPTION
**Pull request by g1itch**
In this branch I'm gonna cover by tests all the crypto functions used in code and do a refactoring. Basic points:

 * only `highlevelcrypto` should import from `pyelliptic`
 * `highlevelcrypto` should not depend on the app state e.g. module `state` or `BMConfigparser()` instance (the same will be done for `protocol` in #1788)
 * `highlevelcrypto` should include hashes - double sha512, ripemd160
 * finally I'm gonna detach `pyelliptic` into the separate repo to show a possible way of addressing #886

### CODE REVIEW

File: src/tests/test_addresses.py [hunk 1]
```diff
@@ -33,3 +35,26 @@ def test_encode(self):
         self.assertEqual(
             sample_address,
             addresses.encodeAddress(2, 1, unhexlify(sample_ripe)))
+
+    def test_wif(self):
+        """Decode WIFs of [chan] bitmessage and check the keys"""
+        self.assertEqual(
+            sample_wif_privsigningkey,
+            highlevelcrypto.decodeWalletImportFormat(
+                b'5K42shDERM5g7Kbi3JT5vsAWpXMqRhWZpX835M2pdSoqQQpJMYm'))
```
Comment (PeterSurda) [inline src/tests/test_addresses.py:44]: these binary data should also be a variable in samples.

File: src/tests/test_crypto.py [hunk 2]
```diff
@@ -62,6 +64,32 @@ def _hashdigest(data):
...
 
+    # def test_sign(self):
+    #     """Check the signature of the sample_msg created with sample key"""
+    #     self.assertEqual(
+    #         highlevelcrypto.sign(sample_msg, sample_privsigningkey), sample_sig)
+
+    def test_double_sha512(self):
+        """Reproduce the example on page 1 of the Specification"""
+        self.assertEqual(
+            highlevelcrypto.double_sha512(b'hello'), sample_double_sha512)
```
Comment (PeterSurda) [inline src/tests/test_crypto.py:75]: ideally `b'hello'` should also be a variable in samples

**Comment by PeterSurda** [comment]
overall nice work, although I haven't investigated it in depth

File: src/tests/test_crypto.py (hunk 2 above)
Comment (g1itch) [reply src/tests/test_crypto.py:75]: It's from here: https://pybitmessage-test.readthedocs.io/en/doc/protocol.html#hashes

**Comment by g1itch** [comment]
> overall nice work, although I haven't investigated it in depth

It seems that I misplaced (or misnamed) the signatures. I was going to regenerate them. And there is also a lot of unfinished changes here, mentioned in the PR description.

File: src/highlevelcrypto.py [hunk 3]
```diff
@@ -7,17 +7,85 @@
...
+    priv = randomBytes(32)
+    pub = pointMult(priv)
+    return priv, pub
+
+
+def deterministic_keys(passphrase, nonce):
+    """Generate keys from *passphrase* and *nonce* (encoded as varint)"""
+    priv = hashlib.sha512(passphrase + nonce).digest()[:32]
+    pub = pointMult(priv)
+    return priv, pub
```
Comment (g1itch) [inline src/highlevelcrypto.py:86]: I'm not sure if these functions should return the pair or only the private key.

**Pull request closed without merging**